master/internal/db/postgres_trial_metrics.go

```diff
@@ -24,6 +26,71 @@ const (
 	GenericMetric MetricPartitionType = "GENERIC"
 )
 
+type metricsBody struct {
+	BatchMetrics interface{}
+	AvgMetrics   *structpb.Struct
+	Type         model.MetricType
+}
+
+func (b metricsBody) ToJSONObj() *model.JSONObj {
+	// we should probably move to avoid special casing based on metric type here.
+	metricsJSONPath := model.TrialMetricsJSONPath(b.Type == model.ValidationMetricType)
+	body := model.JSONObj{
+		metricsJSONPath: b.AvgMetrics,
+	}
+
+	if b.Type == model.ValidationMetricType {
+		return &body
+	}
+
+	// DISCUSS: we can skip this but it'd be a change in behavior.
```

Comment:
it seems if we skip it, we'll delete batch metrics on update? I don't think we can do that.

Reply:
The surrounding condition around the comment has changed. What I meant to convey is that we can probably exclude this if batchMetrics is the zero value to be more consistent with what we do with validation metrics but that'd be a departure from existing behavior.